### PR TITLE
Add GfxIp Register Handler

### PR DESCRIPTION
**Pull request by jiaxie-amd**
- Used for handling[read/store/modify] registers

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit 7d0657670875f938c293e16ae367da456799361c
<details open><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master b601c88aeca7a7b08becb3d32709de383c8ee428</li>
<li>METROHASH: amd-master 2b6fee002db6cc92345b02aeee963ebaaf4c0e2f</li>
<li>PAL: dev e642f608a62887d40d1f25509d2951a4a3576985</li>
<li>SPVGEN: dev d70f9800ce70f89aedb3821f06a69776c58d1d39</li>
<li>XGL: dev 80e5a4b11ad2058097e77746772ddc9ab2118e07</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev b2b163fba8298b8ec6e36cee9666123d0069bfa8</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 960/960 (100.0%)</li>
<li>Failed: 0/960 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.2.0.2-fetch-sources-fix)</summary><ul>


<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 103193/103193 (100.0%)</li>
<li>Failed: 0/103193 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Navi10</summary><ul>

<li>Passed: 103193/103193 (100.0%)</li>
<li>Failed: 0/103193 (0.0%)</li>
</ul></details>


</ul></details>

File: llpc/util/llpcGfxRegHandler.cpp [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+{
+    // mask = ((1 << count) - 1) << offset
+    // Result = (pDword & ~mask)|((pNewBits << offset) & mask)
+    uint32_t mask = ((1 << count) - 1) << offset;
+    Value* pMask = m_pBuilder->getInt32(mask);
+    Value* pNotMask = m_pBuilder->getInt32(~mask);
+    Value* pBeginBit = m_pBuilder->getInt32(offset);
+    pNewBits = m_pBuilder->CreateAnd(m_pBuilder->CreateShl(pNewBits, pBeginBit), pMask);
+    return m_pBuilder->CreateOr(m_pBuilder->CreateAnd(pDword, pNotMask), pNewBits);
+}
```
Comment (nhaehnle) [inline llpc/util/llpcGfxRegHandler.cpp]: Missing newline at end of file

**Comment by jiaxie-amd** [comment]
Fixed.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 935e6a2031a21ab24533aada04a70c30ed773ede
<details open><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master b601c88aeca7a7b08becb3d32709de383c8ee428</li>
<li>METROHASH: amd-master 2b6fee002db6cc92345b02aeee963ebaaf4c0e2f</li>
<li>PAL: dev e642f608a62887d40d1f25509d2951a4a3576985</li>
<li>SPVGEN: dev b30c124e7a042c5251aeb20780a83089355e74a1</li>
<li>XGL: dev 80e5a4b11ad2058097e77746772ddc9ab2118e07</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev b2b163fba8298b8ec6e36cee9666123d0069bfa8</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 960/960 (100.0%)</li>
<li>Failed: 0/960 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.2.0.2-fetch-sources-fix)</summary><ul>


<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 103193/103193 (100.0%)</li>
<li>Failed: 0/103193 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Navi10</summary><ul>

<li>Passed: 103193/103193 (100.0%)</li>
<li>Failed: 0/103193 (0.0%)</li>
</ul></details>


</ul></details>

File: llpc/util/llpcGfxRegHandler.h [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+struct BitsInfo
+{
+    uint32_t index;
+    uint32_t offset;
+    uint32_t count;
+};
+
+// =====================================================================================================================
+// Base class for handling GFX-specific registers
+class GfxIpRegHandlerBase
```
Comment (amdrexu) [inline llpc/util/llpcGfxRegHandler.h]: I forget to point out this: GfxIpRegHandlerBase -> GfxRegHandlerBase

Comment (jiaxie-amd) [reply llpc/util/llpcGfxRegHandler.h]: Okay, I'll  update it soon.

**Comment by amdvlk-admin** [comment]
### Test summary for commit a5842d09c9964a6d621b36599f2a0ad96a16dbf3
<details open><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master b601c88aeca7a7b08becb3d32709de383c8ee428</li>
<li>METROHASH: amd-master 2b6fee002db6cc92345b02aeee963ebaaf4c0e2f</li>
<li>PAL: dev e642f608a62887d40d1f25509d2951a4a3576985</li>
<li>SPVGEN: dev b30c124e7a042c5251aeb20780a83089355e74a1</li>
<li>XGL: dev 80e5a4b11ad2058097e77746772ddc9ab2118e07</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev b2b163fba8298b8ec6e36cee9666123d0069bfa8</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 960/960 (100.0%)</li>
<li>Failed: 0/960 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.2.0.2-fetch-sources-fix)</summary><ul>


<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 103193/103193 (100.0%)</li>
<li>Failed: 0/103193 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Navi10</summary><ul>

<li>Passed: 103193/103193 (100.0%)</li>
<li>Failed: 0/103193 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**